motionplan/tpSpaceRRT.go

```diff
@@ -487,18 +498,37 @@ func (mp *tpSpaceRRTMotionPlanner) checkTraj(trajK []*tpspace.TrajNode, invert b
 
 		sinceLastCollideCheck += math.Abs(trajPt.Dist - lastDist)
 		trajState := &ik.State{Position: spatialmath.Compose(arcStartPose, trajPt.Pose), Frame: mp.frame}
-		nodePose = trajState.Position
 		if sinceLastCollideCheck > mp.planOpts.Resolution {
 			ok, _ := mp.planOpts.CheckStateConstraints(trajState)
 			if !ok {
+				okDist := trajPt.Dist * defaultCollisionWalkbackPct
+				if okDist > defaultMinArcLength && !invert {
+					// Check that okDist is larger than the minimum distance to move to add a partial trajectory.
+					for i := len(passed) - 1; i > 0; i-- {
+						// Return the most recent node whose dist is less than okDist
+						if passed[i].Cost() < okDist {
```

Comment:
This is a little weird.  I think theres a way to get a distance less than the min arc length and also the okDist this way.  I would move the check to see if its bigger than the min arc length here to be sure of it.